Libraries/LibJS/Parser.cpp

```diff
@@ -215,6 +253,53 @@ NonnullOwnPtr<VariableDeclaration> Parser::parse_variable_declaration()
     return make<VariableDeclaration>(make<Identifier>(name), move(initializer), DeclarationType::Var);
 }
 
+NonnullOwnPtr<ForStatement> Parser::parse_for_statement()
+{
+    consume(TokenType::For);
+
+    consume(TokenType::ParenOpen);
+
+    OwnPtr<Statement> init = nullptr;
+    switch (m_current_token.type()) {
+    case TokenType::Var:
+        init = parse_variable_declaration();
```

Comment:
I think `parse_statement` is enough, as `VariableDeclaration` is a statement as well.

Reply:
Oh, good point. I should refer to the spec more.